tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -682,6 +684,173 @@ public void AppWithResources (ApplePlatform platform, string runtimeIdentifiers)
 			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-x64")), File.Exists (x64txt), "x64.txt");
 		}
 
+		[TestCase (ApplePlatform.iOS, "iossimulator-x64")]
+		[TestCase (ApplePlatform.iOS, "ios-arm64;ios-arm")]
+		[TestCase (ApplePlatform.TVOS, "tvossimulator-x64")]
+		[TestCase (ApplePlatform.MacCatalyst, "maccatalyst-x64")]
+		[TestCase (ApplePlatform.MacCatalyst, "maccatalyst-arm64;maccatalyst-x64")]
+		[TestCase (ApplePlatform.MacOSX, "osx-x64")]
+		[TestCase (ApplePlatform.MacOSX, "osx-arm64;osx-x64")] // https://github.com/xamarin/xamarin-macios/issues/12410
+		public void AppWithXCAssets (ApplePlatform platform, string runtimeIdentifiers)
+		{
+			var project = "AppWithXCAssets";
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+
+			var project_path = GetProjectPath (project, runtimeIdentifiers: runtimeIdentifiers, platform: platform, out var appPath);
+
+			DeleteAssets (project_path);
```

Comment:
Are the assets shared between the projects? If they are, the sis going to make it hard to debug.

Reply:
I'm guessing you're deleting and copying assets to avoid duplicating them for all the platforms in every project directory - but does it work if you create a symlink instead? So in each project directory, there would be an Assets.xcassets symlink pointing to the parent folder's TestingAssets directory.

That would make it possible to remove all the delete+copy code.

Reply:
@mandel-macaque  @rolfbjarne 
So my idea behind the copying and deleting was that we could replicate someone adding new assets to a project each time the test runs. I thought this would be good in case the project already included the assets in a previous run and we change something that does not allow them to be included later. Maybe this is not a concern? 

And currently, the directory that they copy from is in `ProjectWithXCAssets/` and they are copying into `ProjectWithXCAssets/{platform}/`. I didn't think this would be an issue, but after talking to @mandel-macaque, there are two testcases with each platform ("iossimulator-x64" and "ios-arm64;ios-arm") which share the `ProjectWithXCAssets/iOS/` project path for example.

I don't know much about symlinks, but I will see if that can work here!

Reply:
> So my idea behind the copying and deleting was that we could replicate someone adding new assets to a project each time the test runs.

That's not a bad idea, but it's a separate test, and would have to be written differently since the tests are by default started from a clean project directory:

1. Build the project
2. Add assets
3. Build again to see what happens.

> Edit: Does my first concern even need to be addressed?

No

> If not then the Symlink idea is just a way to not copy the assets into each project, correct?

Correct, it would just be a way to not quadruplicate all the assets (git is smart and it doesn't matter all that much, but if it works, why not?)